common/lib/xmodule/xmodule/seq_module.py

```diff
@@ -286,10 +287,38 @@ def _get_user(self):
         """
         Return the current runtime Django user.
         """
-        from django.contrib.auth.models import User
-        return User.objects.get(id=self.runtime.user_id)
+        return get_user_model().objects.get(id=self.runtime.user_id)
 
-    def gate_sequence_if_it_is_a_timed_exam_and_contains_content_type_gated_problems(self):
+    def _check_children_for_content_type_gating_paywall(self, item):
+        """
+        If:
+        This xblock contains problems which this user cannot load due to content type gating
+        Then:
+        Return the first content type gating paywall (Fragment)
+        Else:
+        Return None
+        """
+        try:
+            user = self._get_user()
+            course_id = self.runtime.course_id
+            content_type_gating_service = self.runtime.service(self, 'content_type_gating')
+            if not (content_type_gating_service and
+                    content_type_gating_service.enabled_for_enrollment(user=user, course_key=course_id)):
+                self.gated_sequence_paywall = None
+                return None
+
+            for block in traverse_pre_order(item, get_children, leaf_filter):
+                gate_fragment = content_type_gating_service.content_type_gate_for_block(user, block, course_id)
+                if gate_fragment is not None:
+                    # This variable is used for gating timed exams that contain content type gated content
+                    self.gated_sequence_paywall = gate_fragment.content
+                    return gate_fragment
+        except get_user_model().DoesNotExist:
+            pass
+        self.gated_sequence_paywall = None
+        return None
```

Comment:
Nit: you could move the `self.gated_sequence_paywall = None` to the top of the function as the default value that needs to be overridden.

This would also allow you to drop the line above about setting it to None.

Also I would nit on the `return None` at the end of the function, but personal preference.